src/pages/patientView/PatientViewPage.tsx

```diff
@@ -404,6 +436,72 @@ export default class PatientViewPage extends React.Component<
         }
     }
 
+    readonly openResourceTabs = MakeMobxView({
+        await: () => [
+            this.patientViewPageStore.resourceDefinitions,
+            this.patientViewPageStore.resourceIdToResourceData,
+        ],
+        render: () => {
+            const openDefinitions = this.patientViewPageStore.resourceDefinitions.result!.filter(
+                d => this.patientViewPageStore.isResourceTabOpen(d.resourceId)
+            );
+            const sorted = _.sortBy(openDefinitions, d => d.priority);
+            const resourceDataById = this.patientViewPageStore
+                .resourceIdToResourceData.result!;
+
+            const tabs: JSX.Element[] = [];
+            sorted.forEach(def => {
+                const data = resourceDataById[def.resourceId];
+                if (data && data.length > 0) {
+                    tabs.push(
+                        <MSKTab
+                            key={getPatientViewResourceTabId(def.resourceId)}
+                            id={getPatientViewResourceTabId(def.resourceId)}
+                            linkText={def.displayName}
+                            onClickClose={this.closeResourceTab}
+                        >
+                            <OpenResourceTab
+                                resourceData={resourceDataById[def.resourceId]}
+                                urlWrapper={this.urlWrapper}
+                            />
+                        </MSKTab>
+                    );
+                }
+            });
+            return tabs;
+        },
+    });
+
+    @autobind
+    @action
+    private openResource(resource: ResourceData) {
+        // open tab
+        this.patientViewPageStore.setResourceTabOpen(resource.resourceId, true);
+        // go to tab
+        this.urlWrapper.setTab(
```

Comment:
i think this needs comment explaining why these two things are necessary (setTab and then setResourceUrl)